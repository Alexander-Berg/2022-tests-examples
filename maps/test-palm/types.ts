import {Page} from 'puppeteer';

enum Platform {
    DESKTOP = 'desktop'
};

type BrowserCommands = Omit<
    Page,
    | 'executionContext'
    | 'desiredCapabilities'
    | 'sessionId'
    | 'options'
    | 'isIOS'
    | 'isAndroid'
    | 'isPhone'
    | 'then'
    | 'waitForVisible'
> & {
    waitForVisible: (selector: string, milliseconds?: number, reverse?: boolean) => Promise<null>;
};

type TestCommandName = keyof BrowserCommands;

type BrowserStub<T> = {
    [K in TestCommandName]?:
        | null
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        | ((...args: BrowserCommands[K] extends (...args: any) => any ? Parameters<BrowserCommands[K]> : never[]) => T);
} & {
    isPhone: boolean;
};

type GetBrowserStub<T, A extends Record<string, unknown> = {}> = (platform: Platform) => BrowserStub<T> & A;

export {Platform, BrowserStub, GetBrowserStub};
