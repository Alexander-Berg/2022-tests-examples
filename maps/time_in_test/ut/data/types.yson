{
    "id"="type-1";
    "key"="bug";
    "version"=13452345;
};
{
    "id"="type-2";
    "key"="task";
    "version"=13452345;
};
{
    "id"="type-3";
    "key"="feature";
    "version"=13452345;
};
