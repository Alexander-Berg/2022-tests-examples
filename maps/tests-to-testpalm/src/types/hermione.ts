/// <reference types='hermione' />

interface ExcludeReason {
    (reason?: string): never;

    in: (match: RegExp | string, reason: string) => never;
}

interface HermioneContext {
    browser: WebdriverIO.Browser;
    skip: ExcludeReason;
    only: ExcludeReason;
}

type HermionePerformCommand = (
    this: WebdriverIO.Browser,
    callback: () => void | Promise<void>,
    description: string | null
) => void | Promise<void>;

interface MockExcludeReason {
    in: (match: RegExp | string, reason: string) => void;
}

interface MockHermioneContext {
    skip: MockExcludeReason;
    only: MockExcludeReason;
    testPalm: {
        setComponent: (component: string) => void;
    };
}

export {HermioneContext, HermionePerformCommand, MockHermioneContext};
