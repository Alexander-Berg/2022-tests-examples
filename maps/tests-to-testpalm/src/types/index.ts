import {TestCaseAttributes, TestCaseProperty, TestCase, TestCaseStep} from '@yandex-int/testpalm-api';
import type {MockHermioneContext} from './hermione';

type Without<T, U> = {[P in Exclude<keyof T, keyof U>]?: never};

type XOR<T1, T2> = T1 | T2 extends object ? (Without<T1, T2> & T2) | (Without<T2, T1> & T1) : T1 | T2;

enum Engine {
    HERMIONE = 'hermione',
    PUPPETEER = 'puppeteer'
}

interface BrowserDescriptionStub {
    [key: string]: unknown;
    retrieveCommands: () => string[];
    getCommandsWithoutDescription: () => Set<string>;
}

interface BrowserAttributesStub {
    [key: string]: unknown;
    retrieveAttributes: () => TestCaseAttributes;
}

interface BrowserEstimateStub {
    [key: string]: unknown;
    retrieveEstimate: () => number;
    getCommandsWithoutEstimate: () => Set<string>;
}

interface BrowserPreconditionsStub {
    [key: string]: unknown;
    retrievePreconditionsData: () => string;
}

type Stub = XOR<XOR<XOR<BrowserDescriptionStub, BrowserAttributesStub>, BrowserEstimateStub>, BrowserPreconditionsStub>;

interface SuiteCallback<T = undefined> {
    (this: T): void | Promise<void>;
}

interface TestCallback<T = undefined> {
    (this: T): void | Promise<void>;
}

interface SuiteFunction {
    (title: string, callback: SuiteCallback): void;
    only?: (title: string, callback: SuiteCallback) => void | Promise<void>;
    skip?: (title: string, callback: SuiteCallback) => void | Promise<void>;
}

interface TestFunction {
    (title: string, callback: TestCallback): void;
    only?: (title: string, callback: TestCallback) => void | Promise<void>;
    skip?: (title: string, callback: TestCallback) => void | Promise<void>;
}

type HookFunction = (callback: TestCallback) => void;

type SetComponentCommand = (component: string) => void;

type TestsGlobal = Omit<NodeJS.Global, 'beforeEach' | 'afterEach' | 'describe' | 'it' | 'test'> & {
    describe?: SuiteFunction;
    it?: TestFunction;
    test?: TestFunction;

    beforeEach?: HookFunction;
    afterEach?: HookFunction;
} & XOR<{page: Stub}, {hermione: MockHermioneContext}>;

interface TestCaseMeta {
    attributes?: TestCaseAttributes;
    properties?: TestCaseProperty[];
    isAutotest?: boolean;
}

interface SkipNextData {
    skipped?: boolean;
    skipIn?: string;
    skipReason?: string;
}

interface ItRegisteredData extends TestCaseMeta {
    name: string;
    itCallback: TestCallback;
    beforeEachCallbacks: TestCallback[];
    afterEachCallbacks: TestCallback[];
}

interface DescribeRegisteredData extends SkipNextData {
    name: string;
    its: ItRegisteredData[];
    component: string;
    beforeEachCallback?: TestCallback;
    afterEachCallback?: TestCallback;
}

type LocalTestCase = Partial<TestCase> & {stepsExpects: TestCaseStep[]};

interface LocalTestCasesParseData {
    localTestCases: LocalTestCase[];
    parsingErrors: Error[];
}

type StubValue<T> = ((...args: unknown[]) => T) | boolean | null;

interface BrowserStubs {
    attributes: Partial<Record<string, StubValue<TestCaseAttributes>>>;
    description: Partial<Record<string, StubValue<string>>>;
    estimate: Partial<Record<string, StubValue<number>>>;
    preconditions?: {
        // Use `{{key}}` in tamplate from `browserStub.preconditions`.
        template: string;
    } & Partial<Record<string, string | StubValue<Record<string, string>>>>;
}

interface SelectorDescription {
    description: string;
    screenshot: string | null;
}

interface SelectorsDescription {
    [key: string]: SelectorDescription;
}

type Platform = 'desktop' | 'mobile';

interface Config {
    testPalm: {
        projectId: string;
        token: string;
    };
    browserId: string;
    filePatterns: string[];
    getComponentByFile?: (filePath: string) => string;
    browserStubs: BrowserStubs;
    minEstimate?: number;
    getSelectorsDescriptions: () => SelectorsDescription;
    getTicket: () => Promise<string | undefined>;
    report: {
        folder: string;
        name: string;
    };
    testcop: {
        project: string;
        tool: string;
        branch: string;
    };
    engine: Engine;
    trackerQueue: string;
}

type Mode = 'validation' | 'dry-run' | 'full' | 'pr';

interface RunParams {
    configPath: string;
    mode: Mode;
    issue?: string;
}

export {
    MockHermioneContext,
    TestCaseMeta,
    LocalTestCase,
    TestsGlobal,
    TestCallback,
    SuiteCallback,
    SkipNextData,
    ItRegisteredData,
    DescribeRegisteredData,
    LocalTestCasesParseData,
    BrowserStubs,
    SelectorsDescription,
    SelectorDescription,
    Config,
    Mode,
    RunParams,
    TestCaseAttributes,
    Platform,
    BrowserDescriptionStub,
    BrowserAttributesStub,
    BrowserEstimateStub,
    BrowserPreconditionsStub,
    XOR,
    Stub,
    Engine,
    SuiteFunction,
    TestFunction,
    SetComponentCommand
};
