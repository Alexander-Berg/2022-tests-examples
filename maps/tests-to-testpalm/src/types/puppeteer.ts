type PuppeteerPerformCommand = (callback: () => void | Promise<void>, description?: string) => Promise<void>;

export {PuppeteerPerformCommand};
