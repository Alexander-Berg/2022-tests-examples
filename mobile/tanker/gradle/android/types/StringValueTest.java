package ru.yandex.tanker.gradle.android.types;

import org.fest.assertions.api.Assertions;
import org.junit.Test;

public class StringValueTest {

    @Test
    public void partialEscapeTest() throws Exception {
        assertEscapesAs("\'", "\\'");
        assertEscapesAs("\\'", "\\'");
        assertEscapesAs("Well it's working!", "Well it\\'s working!");
        assertEscapesAs("Well it\\'s working!", "Well it\\'s working!");
        assertEscapesAs("wow'", "wow\\'");
        assertEscapesAs("wow\\'", "wow\\'");
        assertEscapesAs("'m fine", "\\'m fine");
        assertEscapesAs("\\'m fine", "\\'m fine");

        assertEscapesAs("a>b", "a\\>b");
        assertEscapesAs("a\\>b", "a\\>b");

        assertEscapesAs("a<b", "a\\<b");
        assertEscapesAs("a\\<b", "a\\<b");

        assertEscapesAs("a&b", "a\\&b");
        assertEscapesAs("a\\&b", "a\\&b");

        assertEscapesAs("a\"b", "a\\\"b");
        assertEscapesAs("a\\\"b", "a\\\"b");
    }

    private void assertEscapesAs(String in, String res) {
        Assertions.assertThat(StringValue.partialEscape(in)).isEqualTo(res);
    }
}
