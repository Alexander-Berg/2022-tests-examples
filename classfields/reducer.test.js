const { PAGE_LOADING_SUCCESS } = require('../../actionTypes');

const reducer = require('./reducer');

let state;
beforeEach(() => {
    state = reducer(undefined, {});
});

describe('PAGE_LOADING_SUCCESS', () => {
    it('не должен обработать, если нет payload.config', () => {
        const nextState = reducer(state, {
            type: PAGE_LOADING_SUCCESS,
            payload: {
                cookies: {},
            },
        });

        expect(nextState === state).toEqual(true);
    });

    it('не должен обработать, если нет payload.cookies', () => {
        const nextState = reducer(state, {
            type: PAGE_LOADING_SUCCESS,
            payload: {
                config: {},
            },
        });

        expect(nextState === state).toEqual(true);
    });
});
