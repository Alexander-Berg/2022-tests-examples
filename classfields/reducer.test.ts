import type { StateAds, UpdateServerDataAction } from 'auto-core/react/dataDomain/ads/types';

import reducer from './reducer';
import { ADS_UPDATE_SERVER_DATA } from './types';

describe('ADS_UPDATE_SERVER_DATA', () => {
    it('должен смежить direct и перезаписать settings', () => {
        const state: StateAds = {
            isFetching: false,
            data: {
                code: 'index',
                data: {
                    direct: {
                        top: {
                            direct_premium: [],
                        },
                    },
                    rtb: {
                        'top:0': {
                            meta: {
                                settings: {},
                            },
                        },
                    },
                },
                settings: {
                    top: {
                        id: 'top',
                        sources: [],
                    },
                },
                statId: '100',
            },
        };
        const action: UpdateServerDataAction = {
            type: ADS_UPDATE_SERVER_DATA,
            payload: {
                direct: {
                    r1: {
                        direct_premium: [],
                    },
                },
                settings: {
                    r1: {
                        id: 'r1',
                        sources: [],
                    },
                },
            },
        };

        expect(reducer(state, action)).toEqual({
            isFetching: false,
            data: {
                code: 'index',
                data: {
                    direct: {
                        r1: {
                            direct_premium: [],
                        },
                        top: {
                            direct_premium: [],
                        },
                    },
                    rtb: {
                        'top:0': {
                            meta: {
                                settings: {},
                            },
                        },
                    },
                },
                settings: {
                    r1: {
                        id: 'r1',
                        sources: [],
                    },
                },
                statId: '100',
            },
        });
    });
});
