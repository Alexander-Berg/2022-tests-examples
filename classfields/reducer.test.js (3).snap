// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`должен мерджить правильный payload в стейт при диспатче экшена PAGE_LOADING_SUCCESS 1`] = `
Object {
  "allow_photo_reorder": 1,
  "auto_activate_cars_offers": 0,
  "auto_activate_commercial_offers": 0,
  "auto_activate_moto_offers": 0,
  "chat_enabled": 0,
  "foo": "foo",
  "hide_license_plate": 0,
  "overdraft_balance_person_id": 0,
  "overdraft_enabled": 0,
}
`;
